testing.md

```diff
@@ -0,0 +1,125 @@
+  }
+
+  function sum(a, b) {
+    return a + b;
+  }
+
+  const number = 1;
+  const secondNumber = addOne(number);
+  const total = sum(number, secondNumber)
+
+  console.log(total) // returns 3
+```
+
+In this example, our tests assertions would be applied to the `total` varible, rather than the functions themselves. We would consider tests as scenarios, such as: 
+- _If `number` is set to 1, would `total` correctly have a value of `3`?_
+- _What happens when the value of `number` is a not a number?_
+
+
+The below example demonstrates how we can write an intergration test, written using [Mocha](#mocha) and the [Node.js Assert Module]():
+
+```javascript
+const assert = require('assert');
+const sum = require('./sum');
+
+test('adds 1 + 2 to equal 3', () => {
+  const number = 1;
+  const secondNumber = addOne(number);
+  const total = sum(number, secondNumber)
+
+  assert(total, 3);
+});
+```
+
+## Node.js Assert Module
+Node ships with a collection of built in modules out of the box, including it's own assertion module. It has a basic selection of comparison functions and can be coupled with a test runner like [Mocha](#mocha) to provide a lightweight testing solution.
+
+Although possible, using the assert module is not recommended for writing tests for a Node application, due to the limited selection of assertions it offers. More robust assertion libraries like [Chai](#chai), or testing frameworks that supply their own assertion libraries like [Jest](#jest), can provide a more robust testing solution for your application.
+
+For further details on the [Assert](https://nodejs.org/api/assert.html) module, check out the [Node.js documentation](https://nodejs.org/api/) for more information.
+
+
+## Mocking HTTP/S Requests
+Testing HTTP/S requests can be a challenging task, due to the asynchronous nature of sending and recieving data between your application and an external nature. 
+
+I good solution for testing your requests is 
+[Nock](https://github.com/nock/nock). 
+The library is great for mocking and intercepting HTTP/S request and works by overriding Node's `http.request` function. It also overrides `http.ClientRequest` too to take modules that use the http module directly into account.
+
+
+For further details on the [HTTP](https://nodejs.org/api/http.html) and [HTTPS](https://nodejs.org/api/https.html) modules, check out the [Node.js documentation](https://nodejs.org/api/) for more information.
+
+## Testing Frameworks
+Below is a selection of some of the most popular testing frameworks available, all of which can be used to test your Node application:
+
+- [Jest](https://jestjs.io)
+- [Mocha](https://mochajs.org/)
+- [AVA](https://github.com/avajs/ava)
+- [Jasmine](https://jasmine.github.io/)
+
+## Testing Modules
+Below are some useful modules that you can use when testing a Node.js application:
+
+### Assertion
+- [Chai](https://www.chaijs.com/)
+- [Must.js](https://github.com/moll/js-must)
+
+### Mocking, Stubbing, and Spying
```

Comment:
When using Jest, all of this is built in, including the coverage tool (it's just wrapping around nyc)